src/hooks/useSelector.js

```diff
@@ -42,11 +42,9 @@ function useSelectorWithStoreAndSubscription(
     throw err
   }
 
-  useIsomorphicLayoutEffect(() => {
-    latestSelector.current = selector
-    latestSelectedState.current = selectedState
-    latestSubscriptionCallbackError.current = undefined
-  })
+  latestSelector.current = selector
+  latestSelectedState.current = selectedState
+  latestSubscriptionCallbackError.current = undefined
```

Comment:
Updating refs in the effect is intentional for concurrent mode.
Do you think this case can be solved by setting the initial ref values?

```js
  const latestSelector = useRef(selector)
  const latestSelectedState = useRef(selectedState)
```

Reply:
> Updating refs in the effect is intentional for concurrent mode.
> Do you think this case can be solved by setting the initial ref values?
> 
> ```js
>   const latestSelector = useRef(selector)
>   const latestSelectedState = useRef(selectedState)
> ```

Ah, is that so? I'm not yet aware of the interactions with concurrent mode. I'll try to provide a solution which keeps it in the `useIsomorphicLayoutEffect` then.

Unfortunately, I don't think setting initial ref values would do anything in this context. The problem is that this block, which is logically updating the `selectedState` cache to its newest value, is using a potentially stale variable. This can happen long after the initial render phase.

Reply:
> This can happen long after the initial render phase.

I see. I misunderstood the issue then. 😥